lessons_17/homework_17.py

```diff
@@ -0,0 +1,149 @@
+# Task 1:
+"""
+Генератор парних чисел від 0 до N
+"""
+def even_numbers(n):
+    for i in range(0, n + 1, 2):
+        yield i
+
+
+
+# Task 2:
+"""
+Генератор, який генерує послідовність Фібоначчі до певного числа N.
+"""
+
+def fibonacci(n):
+    a, b = 0, 1
+    while a <= n:
+        yield a
+        a, b = b, a + b
+
+# Task 3:
+"""
+Ітератор для зворотного виведення елементів списку.
+"""
+
+class ReverseListIterator:
+    def __init__(self, data):
+        self.data = data
+        self.index = len(data)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.index == 0:
+            raise StopIteration
+        self.index -= 1
+        return self.data[self.index]
+    
+# Task 4:
+"""
+Ітератор парних чисел у діапазоні від 0 до N.
+"""
+
+class EvenIterator:
+    def __init__(self, n):
+        self.n = n
+        self.current = 0
+    
+    def __iter__(self):
+        return self
+    
+    def __next__(self):
+        while self.current <= self.n:
+            if self.current % 2 == 0:
+                val = self.current
+                self.current += 1
+                return val
+            self.current += 1
+        raise StopIteration
+    
+# Task 5:
+"""
+Декоратор логування аргументів та результатів
+"""
+
+def logger(func):
+    def wrapper(*args, **kwargs):
+        print(f"Виклик {func.__name__} з аргументами {args}")
+        result = func(*args, **kwargs)
+        print(f"Результат: {result}")
```

Comment:
```suggestion
        log.info(f"Виклик {func.__name__} з аргументами {args}")
        result = func(*args, **kwargs)
        log.info(f"Результат: {result}")
```
А чому прінт а не лог?